src/main/java/org/cryptomator/cloudaccess/MetadataCachingProviderDecorator.java

```diff
@@ -24,31 +24,31 @@
 
 	private final static int DEFAULT_CACHE_TIMEOUT_SECONDS = 10;
 
-	final Cache<CloudPath, CompletionStage<CloudItemMetadata>> itemMetadataCache;
+	final Cache<CloudPath, CompletionStage<CloudItemMetadata>> cachedItemMetadataRequests;
+	final Cache<Map.Entry<CloudPath, Optional<String>>, CompletionStage<CloudItemList>> cachedItemListRequests;
+
 	final Cache<CloudPath, CompletionStage<Quota>> quotaCache;
+
 	private final CloudProvider delegate;
 
 	public MetadataCachingProviderDecorator(CloudProvider delegate) {
-		this(delegate, Duration.ofSeconds( //
-				Integer.getInteger("org.cryptomator.cloudaccess.metadatacachingprovider.timeoutSeconds", DEFAULT_CACHE_TIMEOUT_SECONDS) //
-		));
+		this(
+				delegate, //
+				Duration.ofSeconds(Integer.getInteger("org.cryptomator.cloudaccess.metadatacachingprovider.timeoutSeconds", DEFAULT_CACHE_TIMEOUT_SECONDS))
+		);
 	}
 
 	public MetadataCachingProviderDecorator(CloudProvider delegate, Duration cacheEntryMaxAge) {
 		this.delegate = delegate;
-		this.itemMetadataCache = CacheBuilder.newBuilder().expireAfterWrite(cacheEntryMaxAge).build();
+		this.cachedItemMetadataRequests = CacheBuilder.newBuilder().build();
+		this.cachedItemListRequests = CacheBuilder.newBuilder().build();
```

Comment:
If there is no expiration policy, there is no point in using Guava's Cache, I guess. Either use a normal ConcurrentHashMap or take a look at Caffeine.

Reply:
I strongly assume that the variant of https://github.com/cryptomator/cloud-access-java/pull/5#discussion_r781257124 is easier and less DRY-code instead of implementing another decorator in case the cloud does not support specific caching and we want to cache the responses for a certain time.